test/framework/filetools.py

```diff
@@ -1551,6 +1551,89 @@ def test_copy_files(self):
         error_pattern = "/toy-0.0.eb exists but is not a directory"
         self.assertErrorRegex(EasyBuildError, error_pattern, ft.copy_files, [bzip2_ec], copied_toy_ec)
 
+        # by default copy_files allows empty input list, but if allow_empty=False then an error is raised
+        ft.copy_files([], self.test_prefix)
+        error_pattern = 'One of more files to copy should be specified!'
+        self.assertErrorRegex(EasyBuildError, error_pattern, ft.copy_files, [], self.test_prefix, allow_empty=False)
+
+        # test special case: copying a single file to a file target via target_single_file=True
+        target = os.path.join(self.test_prefix, 'target')
+        self.assertFalse(os.path.exists(target))
+        ft.copy_files([toy_ec], target, target_single_file=True)
+        self.assertTrue(os.path.exists(target))
+        self.assertTrue(os.path.isfile(target))
+        self.assertEqual(toy_ec_txt, ft.read_file(target))
+
+        ft.remove_file(target)
+
+        # default behaviour is to copy single file list to target *directory*
+        self.assertFalse(os.path.exists(target))
+        ft.copy_files([toy_ec], target)
+        self.assertTrue(os.path.exists(target))
+        self.assertTrue(os.path.isdir(target))
+        copied_toy_ec = os.path.join(target, 'toy-0.0.eb')
```

Comment:
Shouldn't this really use basename(toy_ec) instead of hardcoding toy-0.0.eb?

And the same for all other places that does this?

Reply:
hardcoding in the tests is fine, it's often done to ensure we're not just re-implementing the function we're testing (and making the same mistake on both ends, causing the tests to pass while there's actually a problem)